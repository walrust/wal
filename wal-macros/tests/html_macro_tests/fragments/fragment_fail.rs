use wal_macros::html;

fn main() {
    html! { <> };
    html! { <><> };
    html! { <><div></> };
    html! { <></div></> };
    html! { </> };
    html! { <key="key"></key> };
    html! { <key=></> };
    html! { <key="key1" key="key2"></> };
    html! { <other_attr="attr value"></> };
    html! { <key="key" other_attr="attr value"></> };
    html! { <other_attr="attr value" key="key"></> };
}
