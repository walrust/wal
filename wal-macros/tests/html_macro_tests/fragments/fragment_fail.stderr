error: This opening fragment does not have a coressponding closing fragment. (hint: try adding `</>`)
 --> tests/html_macro_tests/fragments/fragment_fail.rs:4:13
  |
4 |     html! { <> };
  |             ^^

error: This opening fragment does not have a coressponding closing fragment. (hint: try adding `</>`)
 --> tests/html_macro_tests/fragments/fragment_fail.rs:5:15
  |
5 |     html! { <><> };
  |               ^^

error: This closing fragment does not have a corresponding opening fragment. (hint: try adding `<>`)
 --> tests/html_macro_tests/fragments/fragment_fail.rs:6:20
  |
6 |     html! { <><div></> };
  |                    ^^^

error: This closing tag does not have a corresponding opening tag. (hint: try adding `<div>`)
 --> tests/html_macro_tests/fragments/fragment_fail.rs:7:15
  |
7 |     html! { <></div></> };
  |               ^^^^^^

error: This closing fragment does not have a corresponding opening fragment. (hint: try adding `<>`)
 --> tests/html_macro_tests/fragments/fragment_fail.rs:8:13
  |
8 |     html! { </> };
  |             ^^^

error: This closing tag does not have a corresponding opening tag. (hint: try adding `<key>`)
 --> tests/html_macro_tests/fragments/fragment_fail.rs:9:24
  |
9 |     html! { <key="key"></key> };
  |                        ^^^^^^

error: Fragment supports only a single `key` attribute
  --> tests/html_macro_tests/fragments/fragment_fail.rs:11:25
   |
11 |     html! { <key="key1" key="key2"></> };
   |                         ^^^

error: Fragment supports only the `key` attribute
  --> tests/html_macro_tests/fragments/fragment_fail.rs:12:14
   |
12 |     html! { <other_attr="attr value"></> };
   |              ^^^^^^^^^^

error: Fragment supports only a single `key` attribute
  --> tests/html_macro_tests/fragments/fragment_fail.rs:13:24
   |
13 |     html! { <key="key" other_attr="attr value"></> };
   |                        ^^^^^^^^^^

error: Fragment supports only a single `key` attribute
  --> tests/html_macro_tests/fragments/fragment_fail.rs:14:14
   |
14 |     html! { <other_attr="attr value" key="key"></> };
   |              ^^^^^^^^^^
